src/kudu/util/net/net_util.cc

```diff
@@ -126,22 +126,49 @@ size_t HostPort::HashCode() const {
 }
 
 Status HostPort::ParseString(const string& str, uint16_t default_port) {
-  std::pair<string, string> p = strings::Split(str, strings::delimiter::Limit(":", 1));
+  /*
+    The `str` param is supposed to take the following combinations:
+    a) Host:Port
+    b) Host
 
-  // Strip any whitespace from the host.
-  StripWhiteSpace(&p.first);
+    Host could be a hostname (devvm1228.prn2.facebook.com) or an IPv6 address.
+    If it is an IPv6 address, it should be wrapped around square brackets.
+
+    According to RFC 3986, [2001:db8:1f70::999:de8:7648:6e8]:100
+    is the format used for specifying IPv6 addresses with host-port.
+  */
+
+  bool ipv6_addr = strcount(str, ':') > 1;
+
+  string host, port_str;
+  if (ipv6_addr) {
+    std::pair<string, string> p = strings::Split(str, strings::delimiter::Limit("]:", 1));
```

Comment:
maybe use a ternary operator and move p out of the block. that way you only need to change the p.first handling.